server/integration_tests/standalone/adversarial.py

```diff
@@ -0,0 +1,459 @@
+          csv_writer.writerow(result.to_csv_row())
+
+  def run_query(self, query: str) -> Result:
+    result = unknown_result(
+        query, self.server_url + f'/explore#q={urllib.parse.quote_plus(query)}')
+    logging.info("Running: %s", query)
+    if not query:
+      logging.info(result)
+      return result
+
+    resp = None
+    try:
+      resp = requests.post(self.server_url +
+                           f'/api/explore/detect-and-fulfill?q={query}',
+                           json={
+                               'contextHistory': {},
+                               'dc': '',
+                           },
+                           timeout=10)
+    except requests.exceptions.ReadTimeout:
+      result.status = ResultStatus.TIMED_OUT
+      logging.info(result)
+      return result
+
+    if resp.status_code != 200:
+      result.status = ResultStatus.REQUEST_FAILED
+      logging.info(result)
+      return result
+
+    response_to_result(resp.json(), result=result)
+    logging.info(result)
+    return result
+
+
+def response_to_result(response: dict, result: Result):
+
+  config = response.get("config")
+  if config is None:
+    result.status = ResultStatus.NO_RESULTS
+    return
+
+  # test for place overview tile
+  has_place_overview_tile = False
+
+  # populate result titles
+  tiles = []
+  svs_used = set()
+  for category in config.get("categories", []):
+    for sv_spec in category.get("statVarSpec", {}).values():
+      sv_used = sv_spec.get("statVar", "")
+      if sv_used:
+        svs_used.add(sv_used)
+
+    for block in category.get("blocks", []):
+      for column in block.get("columns", []):
+        tiles = column.get("tiles", [])
+        for tile in tiles:
+          if not has_place_overview_tile and tile.get("type",
+                                                      "") == "PLACE_OVERVIEW":
+            has_place_overview_tile = tile.get("type", "") == "PLACE_OVERVIEW"
+
+          title = tile.get("title", "")
+          if title:
+            result.result_titles.append(title)
+
+  # populate result places
+  for context in response.get("context", []):
+    for place in context.get("places", []):
+      place_name = place.get("name", "")
+      if place_name:
+        result.result_places.append(place_name)
+
+  # populate llm signals
+  debug = response.get("debug", {})
+  result.llm_detection_type = debug.get("detection_type", "")
+  llm_response = debug.get("query_detection_debug_logs", {}).get("llm_response")
+  if llm_response:
+    result.llm_response = json.dumps(llm_response, indent=1)
+  if svs_used:
+    sv_matching = debug.get("sv_matching", {})
+    cosine_scores = sv_matching.get("CosineScore", [])
+    svs = sv_matching.get("SV", [])
+    if cosine_scores and svs:
+      sv_scores = []
+      for sv, score in zip(svs, cosine_scores):
+        if sv in svs_used:
+          sv_scores.append(f"{sv} = {score}")
+      result.sv_matching = "\n".join(sv_scores)
+
+  # set status
```

Comment:
Add "blocked" and debug["counter"]["INFO"]["info_palm_api_response"].

The latter is the raw response from palm.

Reply:
Ack. Will add this in a follow-on PR.